### ISSUE TITLE
Multi-threading example slow

### ISSUE THREAD

**Issue by cielcao** [numba/numba#4122]
## Reporting a bug

- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

Environment: Numba 0.43.1 with Python 3.7.3 on Mac. 

I am running the example 1.18.3. in https://numba.pydata.org/numba-doc/dev/user/examples.html#multi-threading. The results are

```
numpy (1 thread)         9 ms
numba (1 thread)        49 ms
numba (4 threads)       20 ms
```

I also tried this
```python
@jit(nopython=True)
def func_np_nb(a, b):
    return np.exp(2.1 * a + 3.2 * b)
```

It is as slow as the inner_func_nb one. I also tried on my ubuntu machine and get the same result.

**Comment by esc** [comment]
@cielcao thanks for using Numba and for reporting this. I tried it on my mac and I get the following:

```
💥 zsh» python threading_example.py
numpy (1 thread)        16 ms
numba (1 thread)        37 ms
numba (4 threads)       18 ms
```

**Comment by seibert** [comment]
One possible cause of this is that Anaconda ships with an MKL-accelerated NumPy package.  That package run many of the common math ufuncs (like multiplication, addition, and `exp()`, in this example) automatically multithreaded and SIMD-accelerated.  (That is your "numpy (1 thread)" case might have only 1 python thread, but many MKL threads.)

My assumption is that the example function above is bottlenecked on the computation of `exp()` on an array, which the Anaconda NumPy+MKL will likely be able to run faster than Numba.  If the above code is indicative of your operation, and the size of input array you expect, sticking with NumPy is probably fine.  Numba is generally more beneficial on more complex functions.

(You might be able to get Numba to close the gap with MKL-accelerated ufuncs by installing SVML into your environment and enabling fastmath: https://numba.pydata.org/numba-doc/dev/user/performance-tips.html#intel-svml)

**Comment by cielcao** [comment]
Thanks for the prompt reply. I installed the icc_rt and it did make a difference. I change the size to 10**7 to signify the effect.

without icc_rt installed
'''
numpy (1 thread)       296 ms
numba_np (1 thread)       485 ms
numba (1 thread)       584 ms
numba (4 threads)      178 ms
'''

with icc_rt installed.
'''
numpy (1 thread)       293 ms
numba_np (1 thread)       212 ms
numba (1 thread)       558 ms
numba (4 threads)      168 ms
'''

numba_np is for the func_np_nb I wrote in the previous post, which just add decoration to the numpy function instead of using loop. It seems like the looping version does not get optimized.

By the way this is not related to my project, just got curious about the result :).
